### PR TITLE
Add Fruit_Into_Baskets Implementation 

### LINKED ISSUE #826: [Feature Request] Add Leetcode Fruit Into Baskets Problem Solution
### Is your feature request related to a problem? Please describe.

The repository currently lacks a problem that addresses the "Fruit Into Baskets" challenge, which is an excellent example of using sliding window techniques in algorithmic problem-solving. Adding this problem will enhance the variety of challenges available for contributors.

### Describe the solution you'd like

I propose adding the **Fruit Into Baskets** problem, which involves finding the maximum number of fruits you can collect from trees arranged in a row, given that you can only carry two types of fruits at once. The problem will require contributors to implement a function that utilizes a sliding window approach to efficiently calculate the result.

### Describe alternatives you've considered
N/A

### Additional context
This contribution is part of my participation in GirlScript Summer of Code (GSSoC) 2024 and Hacktoberfest 2024. Adding these algorithms will expand the repository's resources and benefit learners interested in algorithms.

---

**Would you like to work on this feature?**
- [x] Yes

### PR DESCRIPTION
**Pull request by Mahak-Codes**
### Description
This pull request implements the "Fruit into Basket" problem. The task is to collect the maximum number of fruits in two baskets, where each basket can hold only one type of fruit at a time. The goal is to find the longest subarray that contains at most two distinct types of fruits.

Fixes #826

### Type of change

- [x] New feature (non-breaking change which adds functionality)

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: docs/leetcode Solutions/Fruit_Into_Baskets.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
+# Fruit Into Baskets ( Leetcode-904)
```
Comment (ajay-dhangar) [inline docs/leetcode Solutions/Fruit_Into_Baskets.md:9]: We need on top

```md
---
id: ___
title: ___
sidebar_label: ___
sidebar_position: ___
description: ___
tags: [___, __, ]
---
```

### Detailed Guidelines

- id: Unique identifier for the route.
- title: Title of the problem.
- sidebar_label: Problem number or label for the sidebar.
- tags: Relevant tags for the problem.
- description: A short SEO-friendly description. No HTML tags.

File: docs/leetcode Solutions/Fruit_Into_Baskets.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        return maxFruits;
+    }
+};
+
+int main() {
+    Solution solution;
+    vector<int> fruits = {1, 2, 1};
+    cout << "Maximum fruits collected: " << solution.totalFruit(fruits) << endl;
+    return 0;
+}
```
Comment (ajay-dhangar) [inline docs/leetcode Solutions/Fruit_Into_Baskets.md:72]: please close your code example with  ```` ``` ````

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

**Comment by Mahak-Codes** [comment]
@ajay-dhangar I successfully update my code based on suggestions. Can you please review again.

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**